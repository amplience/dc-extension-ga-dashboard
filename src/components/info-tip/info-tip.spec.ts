import { render, fireEvent, screen } from '@testing-library/svelte';
import InfoTip from './info-tip.svelte';

describe('InfoTip', () => {
  it('should render the InfoTip intial state', async () => {
    const { container } = render(InfoTip, {
      content: 'info tip content',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the InfoTip after click', async () => {
    render(InfoTip, {
      content: 'info tip content',
    });

    const element = screen.getByTestId('info-tip');
    await fireEvent.click(element);
    expect(document.body).toMatchSnapshot();
  });
});
