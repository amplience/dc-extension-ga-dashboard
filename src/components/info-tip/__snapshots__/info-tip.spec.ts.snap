// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTip should render the InfoTip after click 1`] = `
<body>
  <div>
    <div
      aria-expanded="false"
      class="svelte-usl6ua"
      data-testid="info-tip"
    >
      <div
        class=" svelte-zpbvvv"
        style="--icon-width: 24px; --icon-height: 24px;"
      >
        <object
          src="ic-asset-info.svg/"
        />
      </div>
    </div>
  </div>
  <div
    data-tippy-root=""
    id="tippy-2"
    style="z-index: 9999; visibility: visible; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 10px);"
  >
    <div
      class="tippy-box"
      data-animation="fade"
      data-escaped=""
      data-placement="bottom"
      data-reference-hidden=""
      data-state="hidden"
      data-theme="amplience"
      role="tooltip"
      style="max-width: 350px;"
      tabindex="-1"
    >
      <div
        class="tippy-content"
        data-state="hidden"
      >
        info tip content
      </div>
      <div
        class="tippy-arrow"
        style="position: absolute; left: 0px; transform: translate(3px, 0px);"
      />
    </div>
  </div>
</body>
`;

exports[`InfoTip should render the InfoTip intial state 1`] = `
<div>
  <div
    aria-expanded="false"
    class="svelte-usl6ua"
    data-testid="info-tip"
  >
    <div
      class=" svelte-zpbvvv"
      style="--icon-width: 24px; --icon-height: 24px;"
    >
      <object
        src="ic-asset-info.svg/"
      />
    </div>
  </div>
</div>
`;
